managed-ledger/src/main/java/org/apache/bookkeeper/mledger/Entry.java

```diff
@@ -65,4 +65,39 @@
      */
     boolean release();
 
+    /**
+     * Get number of batch messages into the entry.
+     * @return
+     */
+    int getNumberOfBatchMessages();
+
+    /**
+     * Set number of batch messages into the entry.
+     * @param numberOfBatchMessages
+     */
+    void setNumberOfBatchMessages(int numberOfBatchMessages);
+
+    /**
+     * Set data size of the entry.
+     * @param totalSizeInBytes
+     */
+    void setTotalSizeInBytes(int totalSizeInBytes);
+
+    /**
+     * Set total number of chunked messages in entry.
+     * @param totalChunkedMessages
+     */
+    void setTotalChunkedMessages(int totalChunkedMessages);
+
+    /**
+     * Get total data size of the entry.
+     * @return
+     */
+    int getTotalSizeInBytes();
+
+    /**
+     * Get total number of chuncked messages.
+     * @return
+     */
+    int getTotalChunkedMessages();
```

Comment:
I think we better not add message related concepts to the entry. Maybe we can add a context carrier in the entry. What do you think?

Reply:
Hi @codelipenghui , could you please provide a little more detail on what you mean by a "context carrier"?

Reply:
I agree with @codelipenghui 
it is unfortunate to add these methods here.
we are going to use them (and the underlying fields) only on the Dispatcher.

it would be better (for instance) to have some EntryWithMetatadata object in Dispatcher that wraps the Entry and adds these fields